gfx/drivers_context/wayland_ctx.c

```diff
@@ -506,11 +507,68 @@ static bool gfx_ctx_wl_bind_api(void *data,
    return false;
 }
 
+static void wl_surface_frame_done(void *data, struct wl_callback *cb, uint32_t time)
+{
+   gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
+
+   wl->swap_complete = true;
+
+   /* Destroy this callback */
+   wl_callback_destroy(cb);
+}
+
+static const struct wl_callback_listener wl_surface_frame_listener = { 
+   .done = wl_surface_frame_done,
+};
+
 static void gfx_ctx_wl_swap_buffers(void *data)
 {
 #ifdef HAVE_EGL
    gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
+
+   settings_t *settings           = config_get_ptr();
+   unsigned max_swapchain_images  = settings->uints.video_max_swapchain_images;
+
+   struct wl_callback *cb = wl_surface_frame(wl->surface);
+
+   if (max_swapchain_images <= 2) {
```

Comment:
The brace style of RA is:

```
if ()
{
}
```
I think?

Reply:
That seems to be the case, yes. Corrected.
Thanks!